lib/pre2017.js

```diff
@@ -0,0 +1,130 @@
+const fetch = require('node-fetch')
+const cheerio = require('cheerio')
+const https = require('https')
+
+const PRE_URL = [
+  'https://codein.withgoogle.com/archive/2016',
+  'https://codein.withgoogle.com/archive/2015',
+  'https://www.google-melange.com/archive/gci/2014',
+  'https://www.google-melange.com/archive/gci/2013',
+  'https://www.google-melange.com/archive/gci/2012',
+  'https://www.google-melange.com/archive/gci/2011',
+  'https://www.google-melange.com/archive/gci/2010',
+]
+
+const CODEIN_WIKI_URL = 'https://en.wikipedia.org/wiki/Google_Code-in'
+
+async function fetchText(url) {
+  const agent = new https.Agent({
+    rejectUnauthorized: false,
+  })
+  const res = await fetch(url, {
+    agent,
+  })
+
+  return await res.text()
+}
+
+async function fetchPre2017(year, url) {
+  const htmlResponse = await fetchText(url)
+  const json =
+    year > 2014
+      ? await scrapeCodein(htmlResponse)
+      : await scrapeMelange(htmlResponse, year)
+
+  return json
+}
+
+async function scrapeCodein(html) {
+  const $ = cheerio.load(html)
+  const mainSelector =
+    'section.home__winners > div.fixed-width > ul.no-style-list > li'
+
+  const resultObj = $(mainSelector).map((index, element) => ({
+    name: getListText($, element, 'div.home__winner-box-header > h3 > a')[0],
+    winners: getListText($, element, 'div:first-child > ul > li > strong'),
+    finalists: getListText($, element, 'div:last-child > ul > li'),
+  }))
+
+  return resultObj.get()
+}
+
+async function scrapeMelange(html, year) {
+  const $ = cheerio.load(html)
+
+  if (year < 2012) {
+    const mainSelector = 'div.main'
+
+    const resultObj = $(mainSelector).map((index, element) => ({
+      winners: getListText($, element, 'ul:first-of-type > li'),
+      organizations: getListText($, element, 'ul:last-child > li a'),
+    }))
+
+    return resultObj.get()
+  }
+
+  const mainSelector = 'div.org-winners-card'
+
+  const resultObj = $(mainSelector).map((index, element) => {
+    const orgInfo = {
+      name: getListText($, element, 'div.mdl-card__title > a')[0],
+      winners: getListText($, element, 'ul:first-of-type > li'),
+    }
+
+    if (year === 2014)
+      orgInfo.finalists = getListText($, element, 'ul:last-child > li')
```

Comment:
If you're using a new line after `if` statement, use curly brackets.

```javascript
// good
if (year === 2014) ...

// bad
if (year === 2014)
  ...

// good
if (year === 2014) {
  ...
}
```

Why?

- See [AirBnB JS Guide](https://github.com/airbnb/javascript#blocks--braces).
- Read more about [ASI](https://tc39.github.io/ecma262/#sec-automatic-semicolon-insertion).